extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php

```diff
@@ -13,130 +13,118 @@
+
+	public function __construct( $query, $moduleName ) {
 		parent::__construct( $query, $moduleName, 'au' );
 		$this->showError = true;
 	}
 
 	private function getUsersForGroup() {
 		global $wgMemc, $wgSpecialsDB;
+
+		if ( empty( $this->params['group'] ) ) {
+			return [];
+		}
+
 		wfProfileIn( __METHOD__ );
 
-		$memkey = sprintf( "%s-%s-%d", __METHOD__, implode("-", (array)$this->params['group'] ), $this->mCityId );
+		$memkey = wfSharedMemcKey( __METHOD__, implode( "-", (array) $this->params['group'] ), $this->mCityId );
 		$data = $wgMemc->get( $memkey );
-		if ( empty($data) ) {
-			$db = wfGetDB(DB_SLAVE, array(), $wgSpecialsDB);
 
-			$group = isset($this->params['group'][0]) ? $this->params['group'][0] : '';
+		if ( !is_array( $data ) ) {
+			$db = wfGetDB( DB_SLAVE, array(), $wgSpecialsDB );
 
-			$where = array(
-				'wiki_id' => intval($this->mCityId),
-				"all_groups" . $db->buildLike( $db->anyString(), $group, $db->anyString() )
+			// SUS-2038: build condition for all groups mentioned in the URL
+			$groupsCondition = array_map(
+				function( $group ) use ( $db ) {
+					return "all_groups" . $db->buildLike( $db->anyString(), $group, $db->anyString() );
+				},
+				$this->params['group']
 			);
 
 			$this->profileDBIn();
-			$oRes = $db->select( 
+
+			// e.g. SELECT  user_id,all_groups FROM `events_local_users`  WHERE wiki_id = '177' AND ((all_groups LIKE '%staff%'  OR all_groups LIKE '%sysop%' ))
+			$oRes = $db->select(
 				'events_local_users',
-				array('user_id, all_groups as ug_groups'),	 
-				$where,
+				[
+					'user_id',
+					'all_groups'
+				],
+				[
+					'wiki_id' => $this->mCityId,
+					join( ' OR ', $groupsCondition )
+				],
 				__METHOD__
 			);
-			
-			$data = array();
-			while ($row = $db->fetchObject($oRes)) {
-				$data[$row->user_id] = $row->ug_groups;
-			}
-			$db->freeResult($oRes);
-			$wgMemc->set( $memkey , $data, 300 );			
-			$this->profileDBOut();
-		}
-
-		wfProfileOut( __METHOD__ );
-		return $data;
-	}
-	
-	private function getAllUserGroups() {
-		global $wgMemc;
-		wfProfileIn( __METHOD__ );
-
-		$memkey = __METHOD__;
-		$data = $wgMemc->get( $memkey );
-		if (empty($data) ) {
-			$db = $this->getSharedDB();
-			$where = array();
 
-			$this->profileDBIn();
-			$oRes = $db->select( 
-				'user_groups', 
-				array('ug_user, ug_group'),	 
-				$where,
-				__METHOD__
-			);
-			
-			$data = array();
-			while ($row = $db->fetchObject($oRes)) {
-				$data[$row->ug_user][] = $row->ug_group;
+			$data = [];
+			while ( $row = $db->fetchObject( $oRes ) ) {
+				$data[$row->user_id] = $row->all_groups;
 			}
-			$db->freeResult($oRes);
-			$wgMemc->set( $memkey , $data, 300 );			
+			$db->freeResult( $oRes );
+			$wgMemc->set( $memkey , $data, 300 );
 			$this->profileDBOut();
 		}
 
 		wfProfileOut( __METHOD__ );
 		return $data;
 	}
 
-	private function getUserRegistration($user_id) {
+	private function getUserRegistration( $user_id ) {
 		global $wgMemc, $wgExternalSharedDB;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "$wgMemc" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)